library/src/main/java/com/novoda/downloadmanager/DownloadBatchRequirementRulesImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.novoda.downloadmanager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+final class DownloadBatchRequirementRulesImpl implements DownloadBatchRequirementRules {
```

Comment:
see `RoomBatchDao_Impl`

Reply:
You want the underscore? I think it's characteristic for generated, rather than manually written code. I can throw it in tho' no probs